android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -150,11 +154,14 @@ constructor(
 
     when (event) {
       is MeasureReportEvent.OnSelectMeasure -> {
-        measureReportConfig.value = event.measureReportConfig
+        event.measureReportConfig?.let {
+          measureReportConfigList.clear()
+          measureReportConfigList.addAll(it)
+        }
         event.navController.navigate(
           MeasureReportNavigationScreen.ReportTypeSelector.route +
             NavigationArg.bindArgumentsOf(
-              Pair(NavigationArg.SCREEN_TITLE, event.measureReportConfig.title)
+              Pair(NavigationArg.SCREEN_TITLE, measureReportConfigList.first().module)
```

Comment:
Using the extension `first()` has a potential of causing `IndexOutOfBoundException` resulting to app crash. Maybe the `measureReportConfigList` is never expected to be empty or null but that extra check for whether the list is empty improves the stability of the code.